Rework word-count documentation (#2247)
exercism/problem-specifications: exercises/word-count/instructions.md

# Instructions

Your task is to count how many times each word occurs in a subtitle of a drama.

> The dramas that the subtitles are taken from use only ASCII characters.

Reviewer: ```suggestion
The subtitles from these dramas use only ASCII characters.
```

The characters often speak in casual English, using contractions like _they're_ or _it's_.
Though these contractions come from two words (e.g. _we are_), the contraction (_we're_) is considered a single word.

Words can be separated by any form of punctuation (e.g. ":", "!", or "?") or whitespace (e.g. "\t", "\n", or " ").
The only punctuation that does not separate words is the apostrophe in contractions.

Numbers are considered words.

> So if the subtitles say `It costs 100 dollars.` then `100` will be its own word.

Reviewer: ```suggestion
If the subtitles say `It costs 100 dollars.` then `100` will be its own word.
```

Words are case insensitive.

> For example, the word `you` occurs three times in the following sentence:

Reviewer: _they're_ uses italics on line 8. `you` uses backticks. Are these intentionally different?

Author: No, I completely missed that. Let me take another look at the whole file.
